sim/src/generic_fss_hardware_model.cpp

```diff
@@ -0,0 +1,241 @@
+                    /* Found it... don't need to go through any more items*/
+                    break; 
+                }
+            }
+        }
+        _time_bus.reset(new NosEngine::Client::Bus(_hub, connection_string, time_bus_name));
+        sim_logger->info("Generic_fssHardwareModel::Generic_fssHardwareModel:  Now on time bus named %s.", time_bus_name.c_str());
+
+        /* Get on a protocol bus */
+        /* Note: Initialized defaults in case value not found in config file */
+        std::string spi_name = "NULL"; // Initialize to default in case value not found in config file
+        int chip_select = 0;
+        if (config.get_child_optional("simulator.hardware-model.connections")) 
+        {
+            /* Loop through the connections for hardware model */
+            BOOST_FOREACH(const boost::property_tree::ptree::value_type &v, config.get_child("simulator.hardware-model.connections"))
+            {
+                /* v.second is the child tree (v.first is the name of the child) */
+                if (v.second.get("type", "").compare("spi") == 0)
+                {
+                    /* Configuration found */
+                    spi_name = v.second.get("bus-name", spi_name);
+                    chip_select = v.second.get("chip-select", chip_select);
+                    break;
+                }
+            }
+        }
+        _spi = new SpiSlaveConnection(this, chip_select, connection_string, spi_name);
+        sim_logger->info("Generic_fssHardwareModel::Generic_fssHardwareModel:  Now on SPI bus name %s, chip select %d.", spi_name.c_str(), chip_select);
+    
+        /* Construction complete */
+        sim_logger->info("Generic_fssHardwareModel::Generic_fssHardwareModel:  Construction complete.");
+    }
+
+
+    Generic_fssHardwareModel::~Generic_fssHardwareModel(void)
+    {        
+        /* Close the protocol bus */
+        delete _spi;
+        _spi = nullptr;
+
+        /* Clean up the data provider */
+        delete _generic_fss_dp;
+        _generic_fss_dp = nullptr;
+
+        /* The bus will clean up the time node */
+    }
+
+    /* Automagically set up by the base class to be called */
+    void Generic_fssHardwareModel::command_callback(NosEngine::Common::Message msg)
+    {
+        /* Get the data out of the message */
+        NosEngine::Common::DataBufferOverlay dbf(const_cast<NosEngine::Utility::Buffer&>(msg.buffer));
+        sim_logger->info("Generic_fssHardwareModel::command_callback:  Received command: %s.", dbf.data);
+
+        /* Do something with the data */
+        std::string command = dbf.data;
+        std::string response = "Generic_fssHardwareModel::command_callback:  INVALID COMMAND! (Try HELP)";
+        boost::to_upper(command);
+        if (command.compare("HELP") == 0) 
+        {
+            response = "Generic_fssHardwareModel::command_callback: Valid commands are HELP, ENABLE, DISABLE, or STOP";
+        }
+        else if (command.compare("ENABLE") == 0) 
```

Comment:
Should update to latest template assuming no endline, so: `command.compare(0,0,"ENABLE)`

Reply:
Done for help, enable, disable, and stop commands.